pypff-stubs/pypff/__init__.pyi

```diff
@@ -0,0 +1,24 @@
+"""
+The stubs file for pypff, a python wrapper for libpff.
+
+We base our understanding of the structure of each class through reading the
+class definitions in [1]. 
+
+One peculiarity within libpff that does not translate well into typed Python,
+is the fact that all methods and instance variables are accessible for each particular object.
+When calling a method or fetching the value of an instance variable, there is no guarantee
+whether a value will be returned, or if an exception is raised.
```

Comment:
hmmmmm do you know why this is the case?

Reply:
This is the case since there's not much preventing a user from making unsafe calls. I suppose the same can be said for Python itself. e.g
```
> arr = [1,2]
> arr[3]
>>> IndexError
```

What I wanted to convey here is that the range of errors is not as clearly defined as things are in Java
```
private int func() throws IOException { ... }
```

Reply:
ahh okay lol - I misread this as "we don't know for sure whether an attribute will be defined"

Reply:
> is the fact that all methods and instance variables are accessible for each particular object.

I read the opposite of what this is supposed to be saying I think. Although I'm still not positive what this one means:

```python
One peculiarity within libpff that does not translate well into typed Python,
is the fact that all methods and instance variables are accessible for each particular object.
```

Is this saying that intended-"private" fields are present in the stub?

Reply:
Intended "private" fields is not quite the intention

For example, with `record_entry`, a `value_type` indicating to the user how to decode the specific `record_entry` is given. That being said, we can still call the full range of "decoding" methods.

Take for example `PidTagMemberName` defined in [1]. The `value_type` corresponds with `PT_STRING8`. To properly decode the value of the `record_entry`, we want to fetch the `data_as_string`. That being said, nothing is preventing us from improperly fetching the `data_as_datetime` or another type.


[1] https://learn.microsoft.com/en-us/office/client-developer/outlook/mapi/pidtagmembername-canonical-property

Reply:
> One peculiarity within libpff that does not translate well into typed Python, is the fact that all methods and instance variables are accessible for each particular object.

Ah okay - so this is just saying that there are cases where you need to check a type value to know which method to call. 

> When calling a method or fetching the value of an instance variable, there is no guarantee whether a value will be returned, or if an exception is raised.

And this is just saying that some methods or property accesses could raise exceptions (which I suppose is the case with any Python library code, and is the case for normal Python properties as well?)

Reply:
I can't think of an example with normal Python properties, since those are typically set and always are expected to have a value if set

The peculiarity here is that with `pypff`, the values of the properties are not set but instead dynamically fetched each time

Reply:
> I can't think of an example with normal Python properties, since those are typically set and always are expected to have a value if set
> 
> The peculiarity here is that with `pypff`, the values of the properties are not set but instead dynamically fetched each time

I think this is how Python properties normally work :D

[Property attributes](https://docs.python.org/3/library/functions.html#property) are a way to make functions look like attributes. One use case that comes to mind is lazily-evaluated attributes:

```python
    @property
    def thing(self) -> Thing:
        if not self._thing:
            print('making the thing!')
            self._thing = self.compute_thing()
        return self._thing
```
```
>>> x = my_obj.thing
making the thing!
>>> x = my_obj.thing
>>>
```